### PR TITLE
chore: support cli to fetch private templates

### PR DESCRIPTION
**Pull request by AayushSaini101**
Resolves: #1081

### CODE REVIEW

File: src/commands/generate/fromTemplate.ts [hunk 1]
```diff
@@ -102,8 +102,17 @@ export default class Template extends Command {
     'map-base-url': Flags.string({
       description: 'Maps all schema references from base url to local folder'
     }),
+    'registry.url': Flags.string({
+      default: 'https://registry.npmjs.org',
+      description: 'Specifies the URL of the private registry for fetching templates and dependencies'
+    }),
+    ' ': Flags.string({
```
Comment (Shurtu-gal) [inline src/commands/generate/fromTemplate.ts:109]: Should be registry.auth I suppose?

File: src/commands/generate/fromTemplate.ts [hunk 2]
```diff
@@ -213,14 +227,33 @@ export default class Template extends Command {
...
 
+  // Parsing the url
+  private registryURLParser(input?:string) {
+    if (!input) { return; }
+    const isURL = /^https?:/;
+    if (!isURL.test(input.toLowerCase())) {
+      throw new Error('Invalid --registry-url flag. The param requires a valid http/https url.');
+    }
+  }
+  // No we need at least one parameter to be passed registryAuth and registryToken
```
Comment (Shurtu-gal) [inline src/commands/generate/fromTemplate.ts:249]: Could you please remove these comments.

File: src/commands/generate/fromTemplate.ts [hunk 3]
```diff
@@ -213,14 +227,33 @@ export default class Template extends Command {
...
+    if (!input) { return; }
+    const isURL = /^https?:/;
+    if (!isURL.test(input.toLowerCase())) {
+      throw new Error('Invalid --registry-url flag. The param requires a valid http/https url.');
+    }
+  }
+  // No we need at least one parameter to be passed registryAuth and registryToken 
+
+  private registryValidation(registryUrl?:string, registryAuth?:string, registryToken?:string) {
+    if (registryUrl!=='https://registry.npmjs.org'&&registryAuth&&!registryToken) {
```
Comment (Shurtu-gal) [inline src/commands/generate/fromTemplate.ts:252]: What is it's a public registry other than npm?

**Review by Shurtu-gal — CHANGES_REQUESTED** [review]
Great work @AayushSaini101 !!
Left a few comments.

Also could we follow the flags naming format till now and rename the flags to `registry-url`, `registry-auth` and `registry-token`

File: src/commands/generate/fromTemplate.ts (hunk 3 above)
Comment (AayushSaini101) [reply src/commands/generate/fromTemplate.ts]: We can make a simple request to registry if it returns bad request, we can assume the registry is private

File: src/commands/generate/fromTemplate.ts (hunk 2 above)
Comment (AayushSaini101) [reply src/commands/generate/fromTemplate.ts]: Done @Shurtu-gal thanks

File: src/commands/generate/fromTemplate.ts (hunk 1 above)
Comment (AayushSaini101) [reply src/commands/generate/fromTemplate.ts]: Yes, added @Shurtu-gal

File: src/commands/generate/fromTemplate.ts [hunk 4]
```diff
@@ -213,14 +228,30 @@ export default class Template extends Command {
...
+  private registryURLParser(input?:string) {
+    if (!input) { return; }
+    const isURL = /^https?:/;
+    if (!isURL.test(input.toLowerCase())) {
+      throw new Error('Invalid --registry-url flag. The param requires a valid http/https url.');
+    }
+  }
+  private async registryValidation(registryUrl?:string, registryAuth?:string, registryToken?:string) {
+    const response= await axios.get(registryUrl as string);
+    if (response.status === 400&&!registryAuth&&!registryToken) {
```
Comment (Shurtu-gal) [inline src/commands/generate/fromTemplate.ts:251]: I believe auth error results in `401` error code.

Comment (AayushSaini101) [reply src/commands/generate/fromTemplate.ts]: Done @Shurtu-gal

**Review by Amzani — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -28,6 +28,7 @@
     "@smoya/multi-parser": "^5.0.7",
     "@stoplight/spectral-cli": "6.9.0",
     "ajv": "^8.12.0",
+    "axios": "^1.6.8",
```
[the remark this answers is not in the document]
Comment (AayushSaini101) [reply package.json]: @Amzani Removed axios use node-fetch instead  thanks

[the remark this answers is not in the document]
Comment (AayushSaini101) [reply package.json]: @Shurtu-gal @Amzani Do we need any new changes in this PR ?

**Review by Amzani — APPROVED** [review]
LGTM
created https://github.com/asyncapi/cli/issues/1421

**Review by Souvikns — APPROVED** [review]

**Review by Shurtu-gal — APPROVED** [review]

**Pull request merged**